dashboard/app/models/census/state_cs_offering.rb

```diff
@@ -1502,7 +1503,33 @@ def self.seed_from_csv(state_code, school_year, update, filename, dry_run = fals
   # @param [string] file_extension
   def self.construct_object_key(state_code, school_year, update = 1, file_extension = 'csv')
     update_string = update == 1 ? "" : ".#{update}"
-    "state_cs_offerings/#{state_code}/#{school_year}-#{school_year + 1}#{update_string}.#{file_extension}"
+    "#{STATE_CS_FOLDER}/#{state_code}/#{school_year}-#{school_year + 1}#{update_string}.#{file_extension}"
+  end
+
+  # Deconstructs an object key into multiple variables.
+  # @param [string] object_key - the AWS object name
+  # @return [array] [state_code, school_year, update, extension]
+  def self.deconstruct_object_key(object_key)
+    # "state_cs_offerings/<state_code>/<school_year>-<school_year_end>(.<update>).<file_extension>"
+    _, state_code, name = object_key.split('/')
+    years, update_or_extension, extension = name.split('.')
```

Comment:
I think I contributed the the naming convention that optionally includes an `update` number in the  object key, so I suppose it's karma that I have a little trouble following this logic. Could it be simplified if we strip the required file extension off of the end first before splitting the name?